functions/anonymous-functions.md

```diff
@@ -0,0 +1,198 @@
+  return parseInt(userId);
+});
+
+console.log(parsedUserIds);
+```
+<!-- prettier-ignore-end -->
+
+Our output shows that `parsedUserIds` is indeed an array of numbers:
+
+```
+[ 401, 403, 404, 500, 518 ]
+```
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                                                                                                                      |
+| --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `const parsedUserIds =`                             | The `map` function on the right-hand side of the `=` will eventually evaluate to become an array. We want to store that array in a `const` variable named `parsedUserIds`. |
+| `userIds.map( ... );`                               | Just like `forEach`, we want to call the `map` function on our array, `userIds`.                                                                                           |
+| `userId => { ... }`                                 | `map` takes in a function. Here is an anonymous arrow function with one parameter, `userId`. `userId` will refer to each item in `userIds` as `map` goes through it.       |
+| `return`                                            | **Unlike** the `forEach` loop, **this function must `return` the value that we want** in our new array.                                                                    |
+
+### `forEach` vs. `map`
+
+The `forEach` function is useful when we want to execute an identical action for each item. `map` is useful when we want a result array that is the same length of the original array, and each item in the result array is based on a parallel item in the original array.
+
+## Uses for Anonymous Functions: `reduce`
+
+Let's look at one more useful function, `reduce`. This function has similarities to `forEach` and `map`, but this time we'll practice using the vocabulary _callback function_.
+
+The responsibility of `reduce` is to look through an array and apply some operation to eventually _reduce_ the array into _one value_.
+
+Let's return to our code that calculated our average cash tip, but let's focus on how we get the _sum_ of our tips.
+
+```javascript
+const cashTips = [4, 7, 9, 12, 3, 18, 6];
+let sum = 0;
+
+cashTips.forEach((tip) => {
+  sum += tip;
+});
+```
+
+Our current logic for getting the sum of all tips is:
+
+1. Initialize a `sum` variable at `0`
+1. Iterate through each `tip` in `cashTips`
+1. Increment `sum` by `tip`
+
+Essentially, we're _reducing_ our `cashTips` array into _one value_. This is a great time to refactor!
+
+The `reduce` function takes in a function. _This_ function takes in _two_ arguments:
+
+1. An argument to represent the accumulated value. This will replace our `sum` variable.
+1. One `item` from our array.
```

Comment:
saying it's a great chance to me somehow sounds more grounded in the situation just described. great time sounds like a general observation.
mention that `reduce` does have an optional second param (possibly worth going more into, as it can protect us from empty lists)
I later suggest using sum rather than ultimateTotal, which then it sound weird to say we replace the sum variable here.

```suggestion
Essentially, we're _reducing_ our `cashTips` array into _one value_. This is a great chance to refactor!

The `reduce` function takes in a function (and an optional starting value). _This_ function takes in _two_ arguments:

1. An argument to represent the accumulated value. This will let us avoid updating the `sum` variable each time through our loop.
1. One `item` from our array.
```